deep-learning/src/main/scala/com/microsoft/azure/synapse/ml/onnx/ONNXRuntime.scala

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.onnx
+
+import ai.onnxruntime.OrtException.OrtErrorCode
+import ai.onnxruntime.OrtSession.SessionOptions
+import ai.onnxruntime.OrtSession.SessionOptions.OptLevel
+import ai.onnxruntime._
+import com.microsoft.azure.synapse.ml.core.env.StreamUtilities.using
+import com.microsoft.azure.synapse.ml.onnx.ONNXUtils._
+import org.apache.spark.TaskContext
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql._
+import org.apache.spark.sql.types._
+
+import scala.collection.JavaConverters._
+import scala.jdk.CollectionConverters.mapAsScalaMapConverter
+
+//noinspection ScalaStyle
+private class ClosableIterator[+T](delegate: Iterator[T], cleanup: => Unit) extends Iterator[T] {
```

Comment:
I think i have an implementation of this class already in SpeechToText project, perhaps we should centralize these two in a util

Reply:
I looked, and they aren't similar enough to make this trivial. The one in StT is based off a LinkedBlockingQueue, not a simple Iterator, and has more features. Since this is just existing code that was moved, I'll leave that to some other future exercise. I did move it to core as a util at least.